sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -575,6 +563,95 @@ case class GpuMax(child: Expression) extends GpuAggregateFunction
   }
 }
 
+/**
+ * All decimal processing in Spark has overflow detection as a part of it. Either it replaces
+ * the value with a null in non-ANSI mode, or it throws an exception in ANSI mode. Spark will also
+ * do the processing for larger values as `Decimal` values which are based on `BigDecimal` and have
+ * unbounded precision. So in most cases it is impossible to overflow/underflow so much that an
+ * incorrect value is returned. Spark will just use more and more memory to hold the value and
+ * then check for overflow at some point when the result needs to be turned back into a 128-bit
+ * value.
+ *
+ * We cannot do the same thing. Instead we take three strategies to detect overflow.
+ *
+ * 1. For decimal values with a precision of 8 or under we follow Spark and do the SUM
+ *    on the unscaled value as a long, and then bit-cast the result back to a Decimal value.
+ *    this means that we can SUM `174,467,442,481` maximum or minimum decimal values with a
+ *    precision of 8 before overflow can no longer be detected. It is much higher for decimal
+ *    values with a smaller precision.
+ * 2. For decimal values with a precision from 9 to 28 inclusive we sum them as 128-bit values.
+ *    this is very similar to what we do in the first strategy. The main differences are that we
+ *    use 128-bit value when doing the sum, and we check for overflow after processing a batch.
+ *    In the case of group-by and reduction that happens after the update stage and also after each
+ *    merge stage. In the worst case this mens that we can SUM `24,028,236,692` maximum or minimum
+ *    decimal values with a precision of 28 before overflow can no longer be detected.
+ * 3. For anything larger than precision 28 we do the same things we do for strategy 2, but we also
+ *    take the digits above 28 and sum them separately. We then check to see if they would have
+ *    overflowed the original limits. This lets us detect overflow in cases where the original
+ *    value would have wrapped around. The reason this works is because we have a hard limit on the
+ *    maximum number of values in a single batch being processed. `Int.MaxValue`, or about 2.2
+ *    billion values. So we use a precision on the higher values that is large enough to handle
+ *    2.2 billion values and still detect overflow. This equates to a precision of about 10 more
+ *    than is needed to hold the higher digits. This effectively gives us unlimited overflow
+ *    detection.
+ *
+ *
+ *
+ * but we do want to have some kind of guarantees that are large
```

Comment:
Not sure if this is a missing capitalization or if a line was accidentally deleted before this one.

Reply:
sorry about that. I didn't finish updating that comment when I was pulled off to other things and ended up forgetting to finish it.